game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/EvaderRetreat.java

```diff
@@ -0,0 +1,143 @@
+      broadcastRetreat(bridge, retreatingPlayer, step, " submerges subs");
+    } else {
+
+      final CompositeChange change = new CompositeChange();
+      change.add(ChangeFactory.moveUnits(battleState.getBattleSite(), retreatTo, units));
+      bridge.addChange(change);
+      battleState.retreatUnits(side, units);
+
+      addHistoryRetreat(bridge, units, " retreated to " + retreatTo.getName());
+      notifyRetreat(battleState, battleActions, units, side, bridge);
+      broadcastRetreat(
+          bridge, retreatingPlayer, step, " retreats", " retreats subs to " + retreatTo.getName());
+    }
+  }
+
+  private static void playSound(
+      final BattleState battleState,
+      final IDelegateBridge bridge,
+      final GamePlayer retreatingPlayer,
+      final Collection<Unit> units) {
+    if (battleState.isHeadless()) {
+      return;
+    }
+    SoundUtils.playRetreatType(retreatingPlayer, units, MustFightBattle.RetreatType.SUBS, bridge);
+  }
+
+  public static void submergeEvaders(
+      final BattleState battleState,
+      final BattleActions battleActions,
+      final Collection<Unit> submerging,
+      final BattleState.Side side,
+      final IDelegateBridge bridge) {
+    final CompositeChange change = new CompositeChange();
+    for (final Unit u : submerging) {
+      change.add(ChangeFactory.unitPropertyChange(u, true, Unit.SUBMERGED));
+    }
+    bridge.addChange(change);
+    battleState.retreatUnits(side, submerging);
+
+    addHistoryRetreat(bridge, submerging, " submerged");
+    notifyRetreat(battleState, battleActions, submerging, side, bridge);
+  }
+
+  private static void addHistoryRetreat(
+      final IDelegateBridge bridge, final Collection<Unit> units, final String suffix) {
+    final String transcriptText = MyFormatter.unitsToText(units) + suffix;
+    bridge.getHistoryWriter().addChildToEvent(transcriptText, new ArrayList<>(units));
+  }
+
+  private static void notifyRetreat(
+      final BattleState battleState,
+      final BattleActions battleActions,
+      final Collection<Unit> retreating,
+      final BattleState.Side side,
+      final IDelegateBridge bridge) {
+    if (battleState.getUnits(side).isEmpty()) {
+      battleActions.endBattle(side.getOpposite().won(), bridge);
+    } else {
+      bridge.getDisplayChannelBroadcaster().notifyRetreat(battleState.getBattleId(), retreating);
+    }
+  }
+
+  private static void broadcastRetreat(
+      final IDelegateBridge bridge,
+      final GamePlayer retreatingPlayer,
+      final String step,
+      final String messageShortSuffix) {
+    broadcastRetreat(bridge, retreatingPlayer, step, messageShortSuffix, messageShortSuffix);
+  }
+
+  private static void broadcastRetreat(
+      final IDelegateBridge bridge,
+      final GamePlayer retreatingPlayer,
+      final String step,
+      final String messageShortSuffix,
+      final String messageLongSuffix) {
+    final String messageShort = retreatingPlayer.getName() + messageShortSuffix;
```

Comment:
I think this method is here to avoid doing this same string concatenation in two places. IMO the indirection with overloaded method args is more harmful than duplicating two lines of code. WDYT of inlining this broadcast method @trevan ?